docs/preset-typescript.md

```diff
@@ -34,10 +34,15 @@ npm install --save-dev @babel/preset-typescript
 
 ### With a configuration file (Recommended)
 
+Note that although TypeScript compiler `tsc` actively supports new JavaScript proposals such as optional chaining (`?.`), nullish coalescing (`??`) and class properties (`this.#x`), this preset does not include these features because they are not
+the types syntax available in TypeScript only. We recommend using `preset-env` with `preset-typescript` if you want to transpile these features.
```

Comment:
The following might be confusing:

> this preset does not include these features because they are not the types syntax available in TypeScript only

Maybe something like:

> this preset does not include these features as it is currently only meant to support the TypeScript types syntax.

Thoughts?